docs/introduction.md

```diff
@@ -67,83 +96,207 @@ Flow is based upon the following set of rules:
+A **block seal** is included in the next block that the consensus nodes finalize, meaning that the block seal for a block is stored in a later block.
+
+### Execution Nodes
+
+The execution nodes determine the results of [transactions](#transactions) when executed in the order determined by the [consensus nodes](#consensus-nodes). 
+They are responsible for scaling the computational power of the blockchain, and are the only node role to have access to the **execution state**.
+Execution nodes follow new blocks as they are published by the consensus nodes.
+The execution process starts when the consensus nodes broadcast evidence that the next block is finalized.
+The process ends when the execution nodes broadcast their **execution receipts**.
+
+Conservative execution nodes only process finalized blocks to avoid wasting resources.
+An execution node can be optimistic — they are allowed to work on unfinalized blocks.
+This can, however, lead to waste or resources if a block is abandoned later.
+Also, validity of unfinalized blocks is not guaranteed — execution node may be processing an invalid block.
 
-External clients submit their transactions to collection nodes. Upon receiving well-formed transactions, collection nodes introduce them to the rest of their cluster.
-The collection nodes of one cluster batch the received transactions into collections. Only a hash reference to a collection is submitted to the consensus nodes for inclusion in a block.
+Execution nodes:
+- receive a finalized block
+- execute the transactions
+- update the execution state of the system
 
-Each cluster of collection nodes generates their collections one at a time. Before a new collection is started, the current one has to be closed and sent to the consensus nodes for inclusion in a block.
-The collection nodes' consensus protocol determines when to start/end a collection and which transactions to include in the collection. The result of that consensus is called a _guaranteed collection_.
+During the execution process the following operations occur:
+- each collection of transactions from the finalized block is retrieved from the relevant cluster
+- execution node executes the transactions according to their canonical order and update the execution state of the system
+- an execution receipt is built for the block. An execution receipt is a cryptographic commitment on the execution result of the block
+- the execution receipt is broadcast to the verification nodes and consensus nodes
 
-##### Verification Nodes
+#### Collection retrieval
+
+Blocks being executed contain a number of guaranteed collections. Guaranteed collections only include a hash of the set of transactions, not their full texts.
+Execution nodes retrieve the collection text from the appropriate cluster, and reconstruct the collection hash from the transaction text to verify data consistency.
+Transactions are processed in order.
```

Comment:
This means that consensus node never have access to the full transactions. That's somewhat important, because it means they can't easily be biased towards including certain transactions over others; at least not at a whole system level. Doesn't stop them from colluding on a actor-by-actor basis.